lib/sts.js

```diff
@@ -1,24 +1,89 @@
-'use strict'
-const aws4 = require('aws4')
-const request = require('request-micro')
+const AWS = require('aws-sdk')
+const crypto = require('crypto')
 
-module.exports = {
-  assumeRole: assumeRole
+function hmac (key, string, encoding) {
+  return crypto
+    .createHmac('sha256', key)
+    .update(string, 'utf8')
+    .digest(encoding)
+}
+
+function hash (string, encoding) {
+  return crypto
+    .createHash('sha256')
+    .update(string, 'utf8')
+    .digest(encoding)
 }
 
-function assumeRole (options) {
-  return request(aws4.sign({
-    service: 'sts',
-    protocol: 'https:',
-    path: `/?Version=2011-06-15&Action=AssumeRole&RoleSessionName=CerberusAssumeRole&RoleArn=${options.assumeRoleArn}`,
-    headers: {
-      'Accept': 'application/json'
-    }
-  })).then(response => {
-    if (!response.statusCode || response.statusCode.toString()[0] !== '2') {
-      throw new Error('Error assuming role. ' + JSON.stringify(response.data.toString()))
-    }
-    return JSON.parse(response.data.toString())
+const getAWSCredentials = () => {
+  const credChain = new AWS.CredentialProviderChain()
+  return new Promise((resolve, reject) => {
+    credChain.resolve((err, credentials) => {
+      if (err) reject(err)
+      resolve(credentials)
+    })
   })
 }
 
+const getAuthenticationHeaders = async (region) => {
+  // Fetch credentials from the AWS Default credentials provider chain
+  let awsCredentials
+  try {
+    awsCredentials = await getAWSCredentials()
+  } catch (e) {
+    throw new Error('Failed to get AWS credentials, do you have IAM credentials available?\n' +
+      'See: https://docs.aws.amazon.com/sdk-for-javascript/v2/developer-guide/setting-credentials-node.html\nmsg: \'' + e.message + '\'')
+  }
+  let { accessKeyId, secretAccessKey, sessionToken } = awsCredentials
+
+  const signedHeaders = 'host;x-amz-date'
+  const date = new Date()
+  const dateISO = date.toISOString().replace(/[:-]|\.\d{3}/g, '')
+  const dateScope = dateISO.slice(0, 8)
+  const dateUTC = date.toUTCString()
+
+  const canonicalRequest = [
+    'POST',
+    '/',
+    '',
+    `host:sts.${region}.amazonaws.com`,
+    `x-amz-date:${dateISO}`,
+    '',
+    signedHeaders,
+    hash('Action=GetCallerIdentity&Version=2011-06-15', 'hex')
+  ].join('\n')
+
+  const canonicalRequestHash = hash(canonicalRequest, 'hex')
+  const stringToSign = [
```

Comment:
We talked about this in person, but I will leave it here in comment form to, so people don't think I ignored you.

There are 2 parts to the answer.

1. Crypto is built into Node so it doesn't need another lib to slow down lambdas
2. I think I was having issues with the Signer from the old SDK, but the new SDK might work, but since this logic is working and doesn't require another lib, we want to leave it as is.